### ISSUE TITLE
some vim apis are unknown to my lsp

### ISSUE THREAD

**Issue by beselig** [nvim-lua/kickstart.nvim#760]
I hope this is worth an issue. If not I apologize, but I don't know where to ask this otherwise.

I have noticed, while going through `init.lua` and following [TJ's video on kickstart](https://www.youtube.com/watch?v=m8C0Cq9Uv9o), that my LSP knows very little about any vim apis. It seems to know some things, like `vim.api` or `vim.lsp`. But the more specific, the higher the chance that I get `unknown` types. For a lot of the commands and functions in the configuration, they definitely work and it's not a problem using them, but the LSP gives me 0 Information about what they are.

It just makes the configuration much harder, because I have no autocomplete on most things.

I have tried upgrading vim, neovim and adding a vim language server. Now I am out of ideas.

## To Reproduce

1. run `brew install neovim`
2. checkout kickstart project as documented
3. hold `K` over vim-specific apis like `vim.lsp.buf.hover` or `vim.api.nvim_create_autocmd`

## Desktop
- OS: MacOS Sonoma 14.2.1
- Terminal: iTerm or terminal

## Neovim Version

```
NVIM v0.9.5
Build type: Release
LuaJIT 2.1.1703358377

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/opt/homebrew/Cellar/neovim/0.9.5/share/nvim"
```

**Comment by dam9000** [comment]
This has been fixed yesterday by #754
Try pulling the latest changes.
On latest master it works properly:
completion:
![image](https://github.com/nvim-lua/kickstart.nvim/assets/148590335/a83cc424-990d-4522-b1fc-13404810331e)
shift-K:
![image](https://github.com/nvim-lua/kickstart.nvim/assets/148590335/46889cae-2e94-4839-9ff9-a0add894468f)

**Comment by beselig** [comment]
Thank you @dam9000. I updated to the prerelease version of neovim and now it works fine.
